src/components/Header/UserPreferences/Automation/index.tsx

```diff
@@ -21,13 +21,13 @@ export default function Automation(): ReactElement {
     useState<boolean>(false)
 
   useEffect(() => {
-    balance.approved &&
+    balance?.approved &&
       setHasZeroERC20TokenBalances(
-        Object.keys(balance.approved)?.filter(
-          (token) => Number(balance.approved[token]) <= 0
+        Object.keys(balance?.approved)?.filter(
+          (token) => Number(balance?.approved[token]) <= 0
```

Comment:
suggestion: since optional chaining on the `balance` object was already added to the above check, it is not necessary here and can be removed.